primefaces/src/main/java/org/primefaces/component/export/ColumnValue.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2009-2023 PrimeTek Informatics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.primefaces.component.export;
+
+public class ColumnValue {
+    private final Object customValue;
+    private final String fallbackValue;
+
+    private ColumnValue(Object customValue, String fallbackValue) {
+        this.customValue = customValue;
+        this.fallbackValue = fallbackValue;
+    }
+
+    public static ColumnValue customValue(Object value) {
+        return new ColumnValue(value, null);
+    }
+
+    public static ColumnValue fallbackValue(String value) {
+        return new ColumnValue(null, value);
+    }
```

Comment:
I want to try the value as custom if the user set it in the attributes `exportValue` or `exportFunction` no matter if is String, LocalDate, BigDecimal or whatever type. So ExcelExporter could handle appropriately, please see my example in https://github.com/primefaces/primefaces/issues/10404#issue-1826639737

Reply:
I understand that, but because you set exportValue or exportFunction doesn't necessarly mean it's a custom value, you might use that just because the column content is not supported by PF, example:

```xml
<p:column exportValue="#{car.color}>
    <div>
          <div>
              #{car.color}
          </div>
    </div>
</p:column>
```

If value is a String, you have to assume it's been properly formatted or formatted in a way that with stronglyTypedCells will try something smart. I understand you want to get rid of that tweak, but this is breaking change, for now stronglyTypedCells can simply be used as a fallback

Reply:
I actually do this in a few places where I may have custom format inside the column but the `exportValue` I use is a different field in the POJO that may be just a String that I want formatted by Strongly Typed Cells.

Reply:
I've already change the name of the variable to "explicitlySetValue" instead of customValue to reflect what I want to mean. The problem with the actual version of PF is when I want to set some column as String and stronglyTypeCell tries to convert to number and strip characters and the excel column has some rows as text and some others as numbers. What do you suggest to solve that without stronglyTypeCell = false?

Reply:
If I understand well, you have a String as input (which look like a number) and is exported as a number: that's what stronglyTypeCell does. Now if you don't want that, then simply disable stronglyTypeCell. There is no bug here so far

Reply:
I'm sorry there is no bug, I mean it's a limitation of stronglyTypedCell. I don't want to disable for entire DataTable just for a single column that it is a String no matter how if looks like number, for example a national ID which starts with "0"

Reply:
So maybe on each column a new `exportStronglyTyped` boolean to allow it to be controlled per column?

Reply:
@jasonex7 @melloware  you can create a map and make custom options per column

Reply:
I think that new attribute in `<p:column>` could be called `exportOptions` and it will be a new interface similar to `org.primefaces.component.export.ExporterOptions` but for columns instead of whole DataTable. That will allow more config in the future. I don't like the map option because it would be too much generic.

Reply:
Not sure what I had in mind at that time... Anyway, just to be clear, from what I understand you're trying to solve 2 issues no? First is, export raw values (as they are and no longer as String) and the other is to have, export options per column right?

Reply:
I want primefaces receive the real value with exportValue or exportFunction properties in <p:colum> and handle that accordingly in exporter. But I understand that there will be a breaking change with String values set in that properties because now it's handle with stronglyTypeCell feature and with this PR as plain String. So in order to not introduce that breaking change, I suggest to add exportOptions property with stronglyTypedCell = true as default to preserve the current behavior.

Reply:
> But I understand that there will be a breaking change with String values set in that properties

No, if a string is set, it's handled as it was before. Why this should be suddenly any different?

Reply:
You're right, I see now what I have to do. I'm going to change this PR that in case stronglyTypedCell = true and the value set was a String then it behave like it was before, any other case is handle by this PR proposal. With that there will be no breaking change.